util/gcs/read_test.go

```diff
@@ -49,6 +49,42 @@ func link(path Path, name, other string) storage.ObjectAttrs {
 	}
 }
 
+func TestBuildJob(t *testing.T) {
+	cases := []struct {
+		path  string
+		build string
+		job   string
+	}{
+		{
+			path:  "gs://bucket/path/job/hello",
```

Comment:
I'm starting to see a pattern here of files used by controller `foo` are generally put into a subfolder called `foo`. I like that organization.